examples/arkode/CXX_parallel/ark_heat2D_lsrk_p.cpp

```diff
@@ -0,0 +1,1740 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ *
+ * (adapted from ark_heat2D_p.cpp, co-authored by Daniel Reynolds and David
+ * Gardner (LLNL))
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Example problem:
+ *
+ * The following test simulates a simple anisotropic 2D heat equation,
+ *
+ *   u_t = kx u_xx + ky u_yy + b,
+ *
+ * for t in [0, 1] and (x,y) in [0, 1]^2, with initial conditions
+ *
+ *   u(0,x,y) = sin^2(pi x) sin^2(pi y),
+ *
+ * stationary boundary conditions
+ *
+ *   u_t(t,0,y) = u_t(t,1,y) = u_t(t,x,0) = u_t(t,x,1) = 0,
+ *
+ * and the heat source
+ *
+ *   b(t,x,y) = -2 pi sin^2(pi x) sin^2(pi y) sin(pi t) cos(pi t)
+ *              - kx 2 pi^2 (cos^2(pi x) - sin^2(pi x)) sin^2(pi y) cos^2(pi t)
+ *              - ky 2 pi^2 (cos^2(pi y) - sin^2(pi y)) sin^2(pi x) cos^2(pi t).
+ *
+ * Under this setup, the problem has the analytical solution
+ *
+ *    u(t,x,y) = sin^2(pi x) sin^2(pi y) cos^2(pi t).
+ *
+ * The spatial derivatives are computed using second-order centered differences,
+ * with the data distributed over nx * ny points on a uniform spatial grid. The
+ * problem is advanced in time with the LSRKStep module in ARKODE.
+ * Several command line options are available to change the problem parameters
+ * and ARKODE settings. Use the flag --help for more information.
+ * ---------------------------------------------------------------------------*/
+
+#include <cmath>
+#include <cstdio>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <sstream>
+
+#include "arkode/arkode_lsrkstep.h"   // access to LSRKStep
+#include "mpi.h"                      // MPI header file
+#include "nvector/nvector_parallel.h" // access to the MPI N_Vector
+#include "sunadaptcontroller/sunadaptcontroller_imexgus.h"
+#include "sunadaptcontroller/sunadaptcontroller_soderlind.h"
+
+// Macros for problem constants
+#define PI    SUN_RCONST(3.141592653589793238462643383279502884197169)
+#define ZERO  SUN_RCONST(0.0)
+#define ONE   SUN_RCONST(1.0)
+#define TWO   SUN_RCONST(2.0)
+#define EIGHT SUN_RCONST(8.0)
+
+// Macro to access (x,y) location in 1D NVector array
+#define IDX(x, y, n) ((n) * (y) + (x))
+
+using namespace std;
+
+// -----------------------------------------------------------------------------
+// User data structure
+// -----------------------------------------------------------------------------
+
+struct UserData
+{
+  // Diffusion coefficients in the x and y directions
+  sunrealtype kx;
```

Comment:
Very inconsistent member naming conventions in this struct

Reply:
These names were all pre-existing in `examples/arkode/CXX_parallel/ark_heat2D_p.cpp`.  For the purposes of an "example" this file should match that one almost identically, except for the items that are _required_ to be different when using LSRKStep vs ARKStep.

If you'd like to update the full suite of `*_heat2D_*` examples then that would be reasonable in a following PR.